### PR TITLE
Enable CES for 1.11+ clusters

### PR DESCRIPTION
**Pull request by Weil0ng**
Enable full CES feature for new 1.11+ clusters.

### CODE REVIEW

File: install/kubernetes/cilium/templates/cilium-configmap.yaml [hunk 1]
```diff
@@ -14,6 +14,7 @@
 {{- $fragmentTracking := "true" -}}
```
Comment (aanm) [inline install/kubernetes/cilium/templates/cilium-configmap.yaml:14]: I've done some local testing and I think the patch should be the following:

```patch
@@ -14,6 +14,7 @@
 {{- $fragmentTracking := "true" -}}
 {{- $crdWaitTimeout := "5m" -}}
 {{- $defaultKubeProxyReplacement := "probe" -}}
+{{- $defaultEnableCiliumEndpointSlice := "false" -}}
 
 {{- /* Default values when 1.8 was initially deployed */ -}}
 {{- if semverCompare ">=1.8" (default "1.8" .Values.upgradeCompatibility) -}}
@@ -37,10 +38,16 @@
 {{- $defaultKubeProxyReplacement = "disabled" -}}
 {{- end -}}
 
+{{- /* Default values when 1.11 was initially deployed */ -}}
+{{- if semverCompare ">=1.11" (default "1.11" .Values.upgradeCompatibility) -}}
+{{- $defaultEnableCiliumEndpointSlice = "true" -}}
+{{- end -}}
+
 {{- $ipam := (coalesce .Values.ipam.mode $defaultIPAM) -}}
 {{- $bpfCtTcpMax := (coalesce .Values.bpf.ctTcpMax $defaultBpfCtTcpMax) -}}
 {{- $bpfCtAnyMax := (coalesce .Values.bpf.ctAnyMax $defaultBpfCtAnyMax) -}}
 {{- $kubeProxyReplacement := (coalesce .Values.kubeProxyReplacement $defaultKubeProxyReplacement) -}}
+{{- $enableCiliumEndpointSlice := (coalesce .Values.enableCiliumEndpointSlice $defaultEnableCiliumEndpointSlice) -}}
 ---
 apiVersion: v1
 kind: ConfigMap
@@ -771,7 +778,7 @@ data:
   vlan-bpf-bypass: {{ .Values.bpf.vlanBypass | join " " | quote }}
 {{- end }}
 
-{{- if .Values.enableCiliumEndpointSlice }}
+{{- if eq $enableCiliumEndpointSlice "true" }}
   enable-cilium-endpoint-slice: "true"
 {{- end }}
```

**Review by aanm — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/templates/cilium-configmap.yaml (hunk 1 above)
Comment (Weil0ng) [reply install/kubernetes/cilium/templates/cilium-configmap.yaml:14]: Thanks @aanm ! Sorry I hadn't spent much time getting this correct, will just shamelessly steal your solution above.

**Review by aanm — APPROVED** [review]

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -2412,6 +2412,9 @@ func (kub *Kubectl) overwriteHelmOptions(options map[string]string) error {
 
 	if CiliumEndpointSliceFeatureEnabled() {
 		options["enableCiliumEndpointSlice"] = "true"
+	} else {
+		// Explicitly unset here because we enable CES in help from 1.11+
```
Comment (kaworu) [inline test/helpers/kubectl.go]:
```suggestion
		// Explicitly unset here because we enable CES in helm from 1.11+
```

**Review by kaworu — APPROVED** [review]

**Review by nbusseneau — APPROVED** [review]

File: test/helpers/kubectl.go (hunk 2 above)
Comment (Weil0ng) [reply test/helpers/kubectl.go]: Oops, done :)

**Comment by Weil0ng** [comment]
Have to change the implementation a bit, test was failing because:
```
executing "cilium/templates/cilium-configmap.yaml" at <eq $enableCiliumEndpointSlice "true">: error calling eq: incompatible types for comparison
```

Turns out `.Values.enableCiliumEndpointSlice` is set to `false`, aka a boolean value, and `defaultEnableCiliumEndpointSlice` is a string value of `"false"`. This results in a type error when doing `eq` for <1.11 installs.

**Comment by pchaigno** [comment]
The net-next CI tests are currently failing because of https://github.com/cilium/cilium/issues/17921.

**Comment by Weil0ng** [comment]
I think this PR is not working as expected, it should only enable CES in CI tests with k8s 1.21+

**Comment by pchaigno** [comment]
> I think this PR is not working as expected, it should only enable CES in CI tests with k8s 1.21+

Setting CI aside for a minute, this PR is enabling CES by default. Does that mean the egress gateway will be broken on default setups? :grimacing:

**Comment by Weil0ng** [comment]
> Setting CI aside for a minute, this PR is enabling CES by default. Does that mean the egress gateway will be broken on default setups? 😬

Ah good point...this is only enabling CES by default for *new* installations, not upgrades, but still yea it breaks egress GW for those new installations...hmm...@aanm WDYT? Should we not enable this by default, maybe we can enable this for new installations in some future 1.11.X? Not sure if changing default is a good practice within the same minor release though...

**Comment by anfernee** [comment]
Or 1.12? Is there a reason for CES to be the default in 1.11.x?

**Comment by Weil0ng** [comment]
> Or 1.12? Is there a reason for CES to be the default in 1.11.x?

Yea, but the hope is that we set it by default asap to get as much as feedback we can.

**Review by pchaigno — CHANGES_REQUESTED** [review]
Are you planning to update this PR to enable CES by default in v1.12+ clusters?

EDIT: Switching to draft mode until the PR is updated or postponed.

**Comment by Weil0ng** [comment]
> Are you planning to update this PR to enable CES by default in v1.12+ clusters?
> 
> EDIT: Switching to draft mode until the PR is updated or postponed.

Yes, I will update this PR for 1.12.

**Comment by Weil0ng** [comment]
This is currently blocked by https://github.com/cilium/cilium/issues/17669, will try to push the fix forward first.

**Pull request closed without merging**